tests/conftest.py

```diff
@@ -33,82 +28,67 @@
 ]
 
 
-def pytest_addoption(parser: _pytest.config.argparsing.Parser) -> None:
-    parser.addoption(
-        "--test_duration",
-        default="short",
-        choices=["short", "long", "all"],
-        help=
-        "Whether to run short tests (the default), long tests, or all tests. A test is considered short if its timeout is "
-        "less than that specified in the config.")
-
-
-def _get_test_world_size(item: pytest.Item) -> int:
-    world_size_marks = tuple(item.iter_markers(name="world_size"))
-    assert len(world_size_marks) <= 1
-    if len(world_size_marks) == 1:
-        world_size_mark = world_size_marks[0]
-        return world_size_mark.args[0]
-    else:
-        return 1
-
-
-def _get_timeout(item: pytest.Item) -> Optional[float]:
-    # see if it has a timeout marker
-    timeout_mark = item.get_closest_marker("timeout")
-    if timeout_mark is None:
-        return None
-    return timeout_mark.args[0]
-
-
-def _filter_items_for_world_size(items: List[pytest.Item]) -> None:
-    items_to_remove: List[pytest.Item] = []
-    env_world_size = int(os.environ.get('WORLD_SIZE', '1'))
+def pytest_addoption(parser: pytest.Parser) -> None:
+    parser.addoption("--duration",
+                     default="short",
+                     choices=["short", "long", "all"],
+                     help="""Duration of tests, one of short, long, or all.
+                             Tests are short if their timeout < 2 seconds
+                             (configurable threshold). Default: short.""")
+    parser.addoption("--world-size",
```

Comment:
This option is just to filter the tests by world_size, and is useful for development purposes (e.g. no world_size envvar set, but would like to know what tests are collecting & running. Since we filter by world_size and duration, prefer to have both of them enumerated at the top rather than being hidden inside one of the functions. 

I clarified the 'help' and also added a guard for then `world_size != os.environ["WORLD_SIZE"]`

Reply:
I see. If this flag is for documentation purposes only (with the validation function below), I can live with that. I guess I would still prefer not to have it unless it serves a functional purpose. (For the collect-only case, I presume one can still filter tests by world size via an env variable -- e.g `WORLD_SIZE=2 pytest --collect-only`)?